core/deployment/src/main/java/io/quarkus/deployment/dev/filesystem/QuarkusFileManager.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkus.deployment.dev.filesystem;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Set;
+
+import javax.tools.ForwardingJavaFileManager;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+
+public abstract class QuarkusFileManager extends ForwardingJavaFileManager<StandardJavaFileManager> {
+
+    protected QuarkusFileManager(StandardJavaFileManager fileManager, Context context) {
+        super(fileManager);
+        try {
+            this.fileManager.setLocation(StandardLocation.CLASS_PATH, context.getClassPath());
+            this.fileManager.setLocation(StandardLocation.CLASS_OUTPUT, List.of(context.getOutputDirectory()));
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot initialize file manager", e);
+        }
+    }
+
+    public abstract Iterable<? extends JavaFileObject> getJavaSources(Iterable<? extends File> files);
+
+    public void reset(Context context) {
```

Comment:
A file manager reset means that if any internal state of the manager needs to be cleared it will be done, i.e. the reloadable file manager needs to be closed before run the next compilation task and the classpath must be set again, but the static file manager is not closed as does not need to scan for external source changes (this is the default behaviour)